controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/DeploymentIssueReporter.java

```diff
@@ -60,35 +62,48 @@ protected void maintain() {
     private void maintainDeploymentIssues(List<Application> applications) {
         List<ApplicationId> failingApplications = new ArrayList<>();
         for (Application application : applications)
-            if (oldApplicationChangeFailuresIn(application.deploymentJobs()))
+            if (oldFailuresIn(application.deploymentJobs(), this::causedByApplicationChange, maxFailureAge))
                 failingApplications.add(application.id());
             else
                 storeIssueId(application.id(), null);
 
         failingApplications.forEach(this::fileDeploymentIssueFor);
 
-        if (controller().versionStatus().version(controller().systemVersion()).confidence() == VespaVersion.Confidence.broken)
-            deploymentIssues.fileUnlessOpen(ApplicationList.from(applications)
-                                                    .upgradingTo(controller().systemVersion())
-                                                    .asList().stream()
-                                                    .map(Application::id)
-                                                    .collect(Collectors.toList()),
-                                            controller().systemVersion());
     }
 
-    private boolean oldApplicationChangeFailuresIn(DeploymentJobs jobs) {
-        if (!jobs.hasFailures()) return false;
+    /**
+     * When the confidence for the system version is BROKEN, file an issue listing the
+     * applications that have been failing the upgrade to the system version for
+     * longer than the set grace period, or update this list if the issue already exists.
+     */
+    private void maintainPlatformIssue(List<Application> applications) {
+        if ( ! (controller().versionStatus().version(controller().systemVersion()).confidence() == VespaVersion.Confidence.broken))
+            return;
+
+        List<ApplicationId> failingApplications = new ArrayList<>();
+        for (Application application : ApplicationList.from(applications).upgradingTo(controller().systemVersion()).asList())
+            if (oldFailuresIn(application.deploymentJobs(), job -> ! causedByApplicationChange(job), upgradeGracePeriod))
```

Comment:
You should put these things in ApplicationList.

Reply:
I hesitated because I didn't think the code would be general and nice at the same time. Take a look and see what you think.